tests/expectations/metrics/test_core.py

```diff
@@ -3975,25 +3963,121 @@ def test_distinct_metric_sa(sa):
     assert pd.Series(index=[1, 2, 3], data=[2, 2, 2]).equals(metrics[desired_metric.id])
     assert pd.Series(index=[4], data=[6]).equals(metrics[desired_metric_b.id])
 
+
+def test_value_counts_metric_pd():
+    engine = build_pandas_engine(pd.DataFrame({"a": [1, 2, 1, 2, 3, 3]}))
+
+    metrics: dict = {}
+
+    table_columns_metric: MetricConfiguration
+    results: dict
+
+    table_columns_metric, results = get_table_columns_metric(engine=engine)
+    metrics.update(results)
+
     desired_metric = MetricConfiguration(
-        metric_name="column.distinct_values",
+        metric_name="column.value_counts",
+        metric_domain_kwargs={"column": "a"},
+        metric_value_kwargs={"sort": "value", "collate": None},
+        metric_dependencies={
+            "table.columns": table_columns_metric,
+        },
+    )
+
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(desired_metric,), metrics=metrics
+    )
+    metrics.update(results)
+    assert pd.Series(index=[1, 2, 3], data=[2, 2, 2]).equals(metrics[desired_metric.id])
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "execution_engine_type",
+    [
+        pytest.param(SqlAlchemyExecutionEngine, id="sa"),
+        pytest.param(SparkDFExecutionEngine, id="spark"),
+    ],
+)
+def test_distinct_metric_sa_and_spark(
+    execution_engine_type: Union[SqlAlchemyExecutionEngine, SparkDFExecutionEngine],
+    sa,
+    spark_session,
+):
+    if execution_engine_type is SqlAlchemyExecutionEngine:
```

Comment:
You could parameterize on this so you don't need to do this condition in the test. Whatever you think is clearer works for me.

Reply:
If you know a better way of parametrizing fixtures, please let us know. We will be doing similar work over the next sprint.